std/container/array.d

```diff
@@ -326,14 +313,22 @@ if (!is(Unqual!T == bool))
             }
             // enlarge
             immutable startEmplace = length;
-            import core.checkedint : mulu;
-            bool overflow;
-            const nbytes = mulu(newLength, T.sizeof, overflow);
-            if (overflow) assert(0);
-            _payload = (cast(T*) realloc(_payload.ptr,
-                            nbytes))[0 .. newLength];
+            if (_capacity < newLength)
+            {
+                import core.checkedint : mulu;
+
+                bool overflow;
+                const nbytes = mulu(newLength, T.sizeof, overflow);
+                if (overflow)
+                    assert(0);
```

Comment:
Personally I don't think this `assert` requires a test, it's only a safety check.